### DISCUSSION TITLE
Preset group:jestPlusTSJest - change match from matchSourceUrlPrefixes to matchPackageNames

### DISCUSSION THREAD

**Discussion by d-koppenhagen** [renovatebot/renovate#17429] [Request Help] [answered]
### What would you like Renovate to be able to do?

The [Group preset `group:jestPlusTSJest`](https://docs.renovatebot.com/presets-group/#groupjestplustsjest) currently matches against a github URL.
This configuration will cause that a dependency is not updated when the source isn't coming from Github (e.g. when having a mirrored registry such as via artifactory).
The matching rule should be changed to match simply the package name.

### If you have any ideas on how this should be implemented, please tell us here.

```diff
{
  "packageRules": [
    {
-      "matchSourceUrlPrefixes": [
-        "https://github.com/kulshekhar/ts-jest"
-      ],
+      "matchPackageNames": [
+        "ts-jest"
+      ],
      "matchUpdateTypes": [
        "major"
      ],
      "groupName": "jest monorepo"
    }
  ]
}
```

### Is this a feature you are interested in implementing yourself?

Maybe

**Comment by viceice** [comment, marked as answer]
We're reading the source url from the package.json, so your artifactory shouldn't strip that information.

`matchSourceUrlPrefixes` doesn't match `registryUrls`, it matches the source repository url from the package.

**Reply by viceice** [reply]
I know it works, as i tested it with verdaccio and proget as npm proxy.

**Reply by rarkins** [reply]
To illustrate what @viceice is saying, load https://registry.npmjs.org/ts-jest and look for the "repository" field. Artifactory should not be modifying that field

**Reply by d-koppenhagen** [reply]
Okay, I was wrong and misinterpreted the field.
Thanks for clarifying!
